Hands-On Workshops – WS 2025/26

# Ana I-2: Reelle Zahlen und Vollständigkeit

***

<div class="alert alert-block alert-danger">
Folgende Begriffe, Konzepte und Sätze sollten am Ende des Sheets verstanden sein:

<ul>
<li> Supremum und Infimum </li>
<li> Maximum und Minimum </li>
<li> Vollständigkeit </li>
<li> Reelle Zahlen </li>
<li> Wurzel </li>
<li> Abzählbarkeit von $\mathbb{Q}$ und Überabzählbarkeit von $\mathbb{R}$
<li> Cantor-Menge </li>
</ul>
</div>

***



Die Vollständigkeit der reellen Zahlen garantiert uns, dass eine nach unten beschränkte Menge $\emptyset\neq M\subseteq\mathbb{R}$ ein Infimum in $\mathbb{R}$ besitzt. Es ist hingegen mitunter nicht einfach, zu entscheiden, ob sogar ein Minimum vorliegt. Ein Beispiel, bei dem kein Minimum vorliegt, haben Sie in der Vorlesung mit $A:=\{x\in\mathbb{R}:x>0\wedge x^2>2\}$ betrachtet. Ähnlich sieht man z. B., dass $\sqrt{5}\notin\mathbb{Q}$ (das kann in der nachfolgenden Aufgabe hilfreich sein und darf ohne weiteren Beweis verwendet werden). Wir schauen uns jetzt ein weiteres Beispiel an.

<div class="alert alert-block alert-success">

### Aufgabe 1
***
Wir betrachten erneut die Fibonacci-Zahlen vom Sheet Ana I-1, d. h. $f_1:=f_2:=1$ und $f_{n+2}:=f_{n+1}+f_n$ für alle $n\in\mathbb{N}$, und betrachten die Menge
\begin{equation*}
M:=\left\{\frac{f_{2n+1}}{f_{2n}}:n\in\mathbb{N}\right\}\subseteq\mathbb{Q}.
\end{equation*}
<ul>
<li> a) <code>[Julia]</code> Berechnen Sie $\frac{f_{2n+1}}{f_{2n}}$ für $n\in\{1,\ldots,20\}$. </li>
<li>b) Begründen Sie, dass $M$ ein Infimum $c\in\mathbb{R}$ besitzt. </li>
<li> c) <code>[Julia]</code> Berechnen Sie $\left(2\cdot\frac{f_{2n+1}}{f_{2n}}-1\right)^2$ für $n\in\{1,\ldots,20\}$ und stellen Sie daraufhin eine Vermutung über den Wert von $c$ auf. </li>
<li> d) Beweisen Sie Ihre Vermutung aus Teilaufgabe c). </li>
<li> e) Besitzt $M$ ein Minimum? </li>
</ul>
</div>

### LÖSUNG AUFGABE 1

In [2]:
function fib(n)

    if n <= 2 
        return 1
    end 

    f_1 = 1
    f_2 = 1

    for i=1:n
        a = f_2 
        f_2 = f_1 + f_2 
        f_1 = a
    end

    return f_2
end 

# a
function m(n)
    return (fib(2*n + 1) / fib(2*n))
end 

for i=1:20
    println(m(i))
end 


5.0
1.625
1.619047619047619
1.6181818181818182
1.6180555555555556
1.6180371352785146
1.618034447821682
1.618034055727554
1.6180339985218033
1.6180339901755971
1.618033988957902
1.6180339887802426
1.6180339887543225
1.6180339887505408
1.618033988749989
1.6180339887499087
1.618033988749897
1.6180339887498951
1.618033988749895
1.618033988749895


b) Begründen Sie, dass $M$ ein Infimum $c\in\mathbb{R}$ besitzt

$M \subseteq \mathbb{Q}$ eine Menge rationaler Zahlen, die ueber eine Rekursion mit positive Naturliche Zahlen mit + definiert 
ist. Also sowohl der Zaehler als auch der Nenner unsere Elemente in $M$ sind positiv und somit $> 0$.

1. Da $M$ nur positive Zahlen enthält, gilt für alle $x \in M$, dass $x > 0$.

2. Daraus folgt, dass $0$ eine untere Schranke von $M$ ist.

3. Da $M$ nach unten beschränkt ist, garantiert Lemma 2.1.21 die Existenz des Infimums von $M$.

$$\therefore \inf(M) \text{ existiert.}$$


 c) <code>[Julia]</code> Berechnen Sie $\left(2\cdot\frac{f_{2n+1}}{f_{2n}}-1\right)^2$ für $n\in\{1,\ldots,20\}$ und stellen Sie daraufhin eine Vermutung über den Wert von $c$ auf. 

In [ ]:
# c = 1 + sqrt(5) / 2 aus 
#
function m2(n)
    return (2 * (fib(2*n + 1) / fib(2*n)) - 1)^2
end 

for i=1:20 
    println(m2(i))
end 

81.0
5.0625
5.009070294784581
5.001322314049586
5.000192901234568
5.00002814344715
5.000004106063733
5.000000599066414
5.000000087402607
5.000000012751869
5.0000000018604736
5.000000000271439
5.000000000039602
5.000000000005778
5.000000000000843
5.0000000000001235
5.000000000000019
5.000000000000003
5.000000000000001
5.000000000000001


d) 

$$\lim_{n \to \infty} \frac{f_{2n + 3}}{f_{2n + 2}} = \lim_{n \to \infty}  \frac{f_{2n + 2} + f_{2n + 1}}{f_{2n + 2}}$$

$$= \lim_{n \to \infty} \left(\frac{f_{2n + 2}}{f_{2n + 2}} + \frac{f_{2n + 1}}{f_{2n + 2}}\right)$$

$$= \lim_{n \to \infty} \left(1 + \frac{f_{2n + 1}}{f_{2n + 1} + f_{2n}}\right)$$

$$= \lim_{n \to \infty} \left(1 + \frac{1}{1 + \frac{f_{2n}}{f_{2n + 1}}}\right)$$

$$= \lim_{n \to \infty} \left(1 + \frac{ 1 }{ 1 + \frac{ 1 }{ \frac{ f_{ 2n + 1} }{ f_{2n} } } }\right)$$

$$= 1 + \frac{1}{\frac{1}{c} + 1} = c$$

$$1 + \frac{1}{\frac{1}{c} + 1} = c$$

$$\frac{1}{c} + 1 = \frac{1+c}{c}$$

$$\frac{1}{\frac{1}{c} + 1} = \frac{c}{1+c}$$

$$1 + \frac{c}{1+c} = c$$

$$\frac{1+c}{1+c} + \frac{c}{1+c} = \frac{1+2c}{1+c}$$

$$\frac{1+2c}{1+c} = c$$

$$1 + 2c = c(1+c) = c + c^2$$

$$1 + 2c - (c + c^2) = 0 \quad \Rightarrow \quad 1 + c - c^2 = 0$$

$$c^2 - c - 1 = 0$$

$$c = \frac{1 \pm \sqrt{1 + 4}}{2} = \frac{1 \pm \sqrt{5}}{2}$$

$$c = \frac{1 + \sqrt{5}}{2} \approx 1.618 \quad \text{or} \quad c = \frac{1 - \sqrt{5}}{2} \approx -0.618$$

e) Nein da Wurzel 5 eine irrationale Zahl ist und unsere Menge besteht nur aus rationale Zahlen. Das heisst, unser Infimum ist nicht in der Menge.

Im Folgenden wollen wir uns etwas genauer mit Wurzeln befassen.

<div class="alert alert-block alert-success">

### Aufgabe 2
***
Bestimmen Sie alle $x\in\mathbb{R}$ mit $x\geq2$, für die $\sqrt{x-2}=x-8$ gilt, und zwar
<ul>
<li> a) <code>[Julia]</code> mit Julia, </li>
<li> b) per Hand. </li>
</ul>
</div>

### LÖSUNG AUFGABE 2

In [2]:
using Pkg; Pkg.add("Polynomials")
using Polynomials

p = Polynomial([66, -17, 1]) 
roots(p)


   Resolving package versions...
     Project No packages added to or removed from `~/.julia/environments/v1.12/Project.toml`
    Manifest No packages added to or removed from `~/.julia/environments/v1.12/Manifest.toml`


2-element Vector{Float64}:
  6.0
 11.0

Wir wollen finden $x \in \mathbb{R}$ so, dass $\sqrt{x - 2} = x - 8$ 

\begin{align*}
   \sqrt{x - 2} &= x - 8 \\ 
    x - 2 &= x^2 -16x + 64 \\
    x^2 - 17x + 66 &= 0 \\ 
    x^2 - 17x + \left(\frac{17}{2}\right)^2 - \left(\frac{17}{2}\right)^2 + 66 = 0 \\ 
    \left(x - \frac{17}{2}\right)^2 = \left(\frac{17}{2}\right)^2 - 66 \\ 
    x - \frac{17}{2} = \sqrt{\left(\frac{17}{2}\right)^2 - 66} \\ 
    x  =  \frac{17}{2} + \sqrt{\left(\frac{17}{2}\right)^2 - 66} \\ 
\end{align*}

$x_1 = 11; x_2 = 6$

Wir kommen nun zur Überabzählbarkeit der Menge der reellen Zahlen - im Gegensatz zur Teilmenge der rationalen Zahlen.

<div class="alert alert-block alert-success">

### Aufgabe 3
***
Im Vorlesungsskript finden Sie mittels der Cantorschen Diagonalverfahren Argumente dahingehend, dass $\mathbb{Q}$ abzählbar und $\mathbb{R}$ überabzählbar ist. Erklären Sie diese Cantorschen Diagonalverfahren. <i>Hinweis: Es reicht, wenn Sie diese Aufgabe mündlich bearbeiten. Kurze Notizen können jedoch hilfreich sein. In jedem Fall wird auf mathematische Präzision Wert gelegt.</i>
</div>

### LÖSUNG AUFGABE 3

Wir wollen uns im Folgenden noch mit der Cantor-Menge $\mathcal{C}\subseteq\mathbb{R}$ beschäftigen. Zunächst beschreiben wir deren Konstruktion informell: Man startet mit dem Einheitsintervall $A_0:=[0,1]$ und entfernt daraus das ,,mittlere Drittel'' (wobei die Randpunkte jeweils nicht mit entfernt werden). Das Resultat $A_1$ besteht folglich aus den Teilintervallen $\left[0,\frac{1}{3}\right]$ und $\left[\frac{2}{3},1\right]$. Nun entfernt man aus diesen beiden Teilintervallen jeweils das mittlere Drittel und fährt immer weiter so fort. Für festes, aber beliebiges $n\in\mathbb{N}_0$ besteht $A_n$ also aus $2^n$ paarweise disjunkten abgeschlossenen Teilintervallen der Länge $3^{-n}$, und $A_{n+1}$ geht aus $A_n$ hervor, indem aus jedem dieser Teilintervalle das mittlere Drittel entfernt wird. Das, was ,,übrig bleibt'', ist die Cantor-Menge. Nun mathematisch präzise: Zu $n\in\mathbb{N}_0$ sei $A_{n+1}:=\frac{1}{3}(A_n\cup(2+A_n))$ (zur Notation - für $X\subseteq\mathbb{R}$ und $a,t\in\mathbb{R}$ ist definiert $a X:=\{a x:x\in X\}$, $t+X:=\{t+x:x\in X\}$). Dann $\mathcal{C}:=\bigcap_{n=0}^\infty A_n$.

<div class="alert alert-block alert-success">

### Aufgabe 4
***

- a) Zeigen Sie, dass $\mathcal{C}$ überabzählbar ist. Hinweis: Nehmen Sie an, es gäbe eine surjektive Abbildung $c:\mathbb{N}\rightarrow\mathcal{C}$. Sei $I_1$ ein Teilintervall von $A_1$ mit $c(1)\notin I_1$, und sei zu $n\in\mathbb{N}$ die Menge $I_{n+1}$ ein Teilintervall von $I_n\cap A_{n+1}$ mit $c(n+1)\notin I_{n+1}$. Zeigen Sie, dass $\bigcap_{n=1}^\infty I_n\neq\emptyset$ und leiten Sie daraus einen Widerspruch ab.

- b) Folgern Sie ohne Verwendung des Cantorschen Diagonalverfahrens, dass $\mathbb{R}$ überabzählbar ist.

- c) Besitzt $\mathcal{C}$ Elemente, die kein Randpunkt eines Teilintervalls einer der Mengen $A_n$ sind? 
<i> Ergänzende Bemerkung: Eine Visualisierung der Cantor-Menge (genauer gesagt der Obermengen $A_n$) finden Sie auf der Mathswell-Website unter folgendem Link: https://mathswell.com/cantor-set/#explorer</i>
</div>

a) 

**Annahme**

Angenommen, die Cantor-Menge sei abzaehlbar. Dann existiert eine bijektive Funktion  

$$
c:\mathbb{N} \to \mathcal{C}.
$$

Nach dem ersten Schritt der Cantor-Konstruktion gilt  

$$
A_1 = [0,\tfrac13] \cup [\tfrac23,1].
$$

Da  

$$
c(1) \in \mathcal{C} \subset A_1,
$$
liegt $c(1)$ in genau einem der beiden disjunkten Intervalle.  
Wir wählen das jeweils andere Intervall und nennen es $I_1$.

Damit gilt:

- $I_1\subset A_1$,
- $I_1$ ist abgeschlossen,
- $c(1)\notin I_1$.


**Induktionsschritt**

Angenommen, es existiert ein abgeschlossenes Intervall  

$$
I_n \subset A_n \quad\text{mit}\quad c(n)\notin I_n.
$$

Beim Uebergang von $A_n$ zu $A_{n+1}$ wird das mittlere Drittel jedes Intervalls entfernt.  
Daher zerfällt auch $I_n$ in genau zwei disjunkte abgeschlossene Teilintervalle von $A_{n+1}$.

Da $c(n+1)$ nur in einem dieser beiden Intervalle liegen kann, waehlen wir das andere und nennen es $I_{n+1}$.

Damit gilt:

- $I_{n+1}\subsetneq I_n$,
- $I_{n+1}$ ist abgeschlossen und nicht leer,
- $c(n+1)\notin I_{n+1}$


Durch die Konstruktion entsteht eine geschachtelte Folge
$$
I_1 \supsetneq I_2 \supsetneq I_3 \supsetneq \cdots
$$

Die Laengen gehen also gegen 0.

Da die Intervalle abgeschlossen, beschraenkt und geschachtelt sind und ihre Laenge gegen 0 geht, folgt aus dem Intervallschachtelungsprinzip:

$$
\bigcap_{n=1}^{\infty} I_n \neq \varnothing.
$$

Es existiert genau ein Punkt  
$$
x \in \bigcap_{n=1}^{\infty} I_n.
$$

Da jedes $I_n \subset A_n$ gilt, folgt  

$$
x \in \mathcal{C}.
$$


**Widerspruch**

Weil $c$ nach Annahme surjektiv ist, existiert ein $k\in\mathbb{N}$ mit  

$$
x = c(k).
$$

Nach unserer Konstruktion wurde jedoch $I_k$ so gewaehlt, dass  
$$
c(k)\notin I_k,
$$

waehrend  
$$
x \in I_k.
$$

Das ist ein Widerspruch.


Die Annahme, dass die Cantor-Menge abzählbar ist, führt zu einem Widerspruch.

b) Die Cantor Menge ist eine Teilmenge der Reelle Zahlen und unabzaehlbar von daher ist sind auch die reelle Zahlen unabzaehlbar
Angenommen es existierte eine Bijektion vom N auf R, dann gabe es eine Bijektion auch auf die Cantor Menge dessen Existenz schon widerlegt wurde, ergo R ist unazaelbar. 

c) Es gibt endlich viele Randwerte und es gibt abzaehlbare  $2^n$ endlich viele Schritte, aber die Cantor Menge ist 
ueberabzaehlbar, deshalb existieren Punkte die keine Rankpunkte sind.